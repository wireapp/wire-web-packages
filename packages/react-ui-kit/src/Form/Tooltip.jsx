/*
 * Wire
 * Copyright (C) 2018 Wire Swiss GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see http://www.gnu.org/licenses/.
 *
 */

import {COLOR} from '../Identity';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Tooltip = styled.div.attrs({'data-text': ({text}) => text})`
  position: relative;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;

  &::after {
    position: absolute;
    content: attr(data-text);
    display: block;
    opacity: 0;
    transform: translateY(16px);
    bottom: calc(100% + 8px);
    ${({bottom}) =>
      bottom &&
      `
      transform: translateY(-16px);
      bottom: auto;
      top: calc(100% + 8px);
    `};
    ${({left}) =>
      left &&
      `
      transform: translateX(16px);
      bottom: auto;
      right: calc(100% + 8px);
    `};
    ${({right}) =>
      right &&
      `
      transform: translateX(-16px);
      bottom: auto;
      right: auto;
      left: calc(100% + 8px);
    `};
    pointer-events: none;
    border-radius: 4px;
    transition: all 0.15s ease-in-out;
    padding: 12px;
    min-width: 120px;
    max-width: 200px;
    text-align: center;
    font-size: 12px;
    line-height: 14px;
    box-shadow: 0 2px 16px 0 rgba(0, 0, 0, 0.12);
    ${({light}) =>
      light
        ? `
    background-color: ${COLOR.WHITE};
    color: ${COLOR.TEXT};
    font-weight: 400;
    `
        : `
    background-color: ${COLOR.TEXT};
    color: ${COLOR.WHITE};
    font-weight: 600;
    `};
  }
  ${({disabled}) =>
    !disabled &&
    `
  &:hover::after {
    transition: all 0.25s ease-in-out;
    opacity: 1;
    transform: translateY(0) translateX(0);
  }`};
`;

Tooltip.propTypes = {
  bottom: PropTypes.bool,
  disabled: PropTypes.bool,
  left: PropTypes.bool,
  right: PropTypes.bool,
  text: PropTypes.string,
};

Tooltip.defaultProps = {
  bottom: false,
  disabled: false,
  left: false,
  right: false,
  text: '',
};

export {Tooltip};
