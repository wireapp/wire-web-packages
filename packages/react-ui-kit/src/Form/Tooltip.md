Demo:

```js
import {Fragment} from 'react';
import {Form, Tooltip, Input} from '@wireapp/react-ui-kit';

<Fragment>
  <Form
    style={{width: 100, display: 'flex', flexDireciton: 'column', margin: '0 auto'}}
    onSubmit={event => {
      event.preventDefault();
    }}
  >
    <Tooltip text="This is a tooltip on top">
      <Input placeholder="Tooltip on top" />
    </Tooltip>
    <Tooltip text="This is a tooltip on right" right>
      <Input placeholder="Tooltip on right" />
    </Tooltip>
    <Tooltip text="This is a tooltip on left" left>
      <Input placeholder="Tooltip on left" />
    </Tooltip>
    <Tooltip text="This is a tooltip on bottom" bottom>
      <Input placeholder="Tooltip on bottom" />
    </Tooltip>
  </Form>
</Fragment>;
```
