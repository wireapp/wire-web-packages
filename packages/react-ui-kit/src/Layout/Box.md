Demo:

```js
import {Box} from '@wireapp/react-ui-kit';

<Box>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce a mattis nibh, sed maximus leo. Fusce a lacinia sem,
  vitae ornare dolor. Quisque rhoncus, magna non lacinia sagittis, erat augue fringilla metus, eu consectetur leo velit
  non lacus. Phasellus ipsum turpis, dapibus ut purus in, lobortis consectetur mi.
</Box>;
```
