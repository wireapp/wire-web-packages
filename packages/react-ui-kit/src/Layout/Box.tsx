/*
 * Wire
 * Copyright (C) 2018 Wire Swiss GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see http://www.gnu.org/licenses/.
 *
 */

/** @jsx jsx */
import {ObjectInterpolation, jsx} from '@emotion/core';
import React from 'react';
import {COLOR} from '../Identity';

export interface BoxProps<T = HTMLDivElement> extends React.HTMLProps<T> {}

const boxStyles: (props: BoxProps) => ObjectInterpolation<undefined> = props => ({
  border: `2px solid ${COLOR.GRAY_LIGHTEN_72}`,
  borderradius: '8px',
  padding: '16px 32px',
});

const Box = (props: BoxProps) => <div css={boxStyles(props)} {...props} />;

export {Box, boxStyles};
