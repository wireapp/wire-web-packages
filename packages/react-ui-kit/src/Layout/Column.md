Demo:

```js
import {Container, Columns, Column, COLOR} from '@wireapp/react-ui-kit';

const ColumnsStyle = {
  marginBottom: '12px',
};

const ColumnStyle = {
  backgroundColor: COLOR.GRAY_LIGHTEN_32,
  border: `1px solid ${COLOR.GRAY_LIGHTEN_24}`,
};

<Container>
  <Columns style={ColumnsStyle}>
    <Column style={ColumnStyle}>Column</Column>
  </Columns>

  <Columns style={ColumnsStyle}>
    <Column style={ColumnStyle}>Column</Column>
    <Column style={ColumnStyle}>Column</Column>
  </Columns>

  <Columns style={ColumnsStyle}>
    <Column style={ColumnStyle}>Column</Column>
    <Column style={ColumnStyle}>Column</Column>
    <Column style={ColumnStyle}>Column</Column>
  </Columns>

  <Columns style={ColumnsStyle}>
    <Column style={ColumnStyle}>Column</Column>
    <Column style={ColumnStyle}>Column</Column>
    <Column style={ColumnStyle}>Column</Column>
    <Column style={ColumnStyle}>Column</Column>
  </Columns>

  <Columns style={ColumnsStyle}>
    <Column style={ColumnStyle}>Column</Column>
    <Column style={ColumnStyle}>Column</Column>
    <Column style={ColumnStyle}>Column</Column>
    <Column style={ColumnStyle}>Column</Column>
    <Column style={ColumnStyle}>Column</Column>
  </Columns>

  <Columns style={ColumnsStyle}>
    <Column style={ColumnStyle}>
      <Columns>
        <Column style={ColumnStyle}>Column</Column>
        <Column style={ColumnStyle}>Column</Column>
        <Column style={ColumnStyle}>Column</Column>
      </Columns>
    </Column>
    <Column style={ColumnStyle}>
      <Columns>
        <Column style={ColumnStyle}>Column</Column>
        <Column style={ColumnStyle}>Column</Column>
        <Column style={ColumnStyle}>Column</Column>
      </Columns>
    </Column>
    <Column style={ColumnStyle}>
      <Columns>
        <Column style={ColumnStyle}>Column</Column>
        <Column style={ColumnStyle}>Column</Column>
        <Column style={ColumnStyle}>Column</Column>
      </Columns>
    </Column>
  </Columns>
</Container>;
```
