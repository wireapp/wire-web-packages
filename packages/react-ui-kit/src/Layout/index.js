import setGlobalStyles from '../globalStyles';

setGlobalStyles();

export * from './Column';
export * from './Container';
export * from './Content';
export * from './Footer';
export * from './Header';
export * from './StyledApp';
