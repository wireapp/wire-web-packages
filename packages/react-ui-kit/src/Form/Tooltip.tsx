/*
 * Wire
 * Copyright (C) 2022 Wire Swiss GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see http://www.gnu.org/licenses/.
 *
 */

import * as React from 'react';

import {CSSObject} from '@emotion/react';

import {Theme} from '../Layout';
import {filterProps} from '../util';

interface ToolTipProps<T = HTMLDivElement> extends React.HTMLProps<T> {
  position?: 'top' | 'right' | 'left' | 'bottom';
  body: React.ReactNode;
  isOpen?: boolean;
}

const tooltipStyle: (theme: Theme) => CSSObject = theme => ({
  position: 'relative',
  width: 'fit-content',
  '&:hover .tooltip-content, &:focus-within .tooltip-content,': {
    visibility: 'visible',
    opacity: 1,
  },
  '.tooltip-content': {
    textAlign: 'center',
    visibility: 'hidden',
    opacity: 0,
    width: 'max-content',
    height: 'max-content',
    position: 'absolute',
    boxSizing: 'border-box',
    display: 'block',
    margin: '0 auto',
    padding: '4px 8px',
    backgroundColor: theme.Tooltip.backgroundColor,
    color: theme.Tooltip.color,
    borderRadius: 4,
    fontSize: theme.fontSizes.small,
    fontWeight: 500,
    whiteSpace: 'nowrap',
    transition: 'opacity ease-out 150ms, bottom ease-out 150ms',
    boxShadow: '1px 2px 6px rgba(0, 0, 0, 0.3)',
  },
  '& .tooltip-content .tooltip-arrow': {
    position: 'absolute',
    width: 0,
    height: 0,
  },
  "&[data-position='top'] .tooltip-content": {
    bottom: '100%',
    marginBottom: 10,
    left: '50%',
    transform: 'translateX(-50%)',
    '& .tooltip-arrow': {
      filter: 'drop-shadow(0px 2px 1px rgba(0, 0, 0, 0.1))',
      borderLeft: '12px solid transparent',
      borderRight: '12px solid transparent',
      borderTop: `12px solid ${theme.Tooltip.backgroundColor}`,
      top: '90%',
      left: '50%',
      transform: 'translateX(-50%)',
    },
  },
  "&[data-position='right'] .tooltip-content": {
    left: '100%',
    marginLeft: '8px !important',
    top: 0,
    bottom: 0,
    margin: 'auto 0',
    '& .tooltip-arrow': {
      filter: 'drop-shadow(-2px 0px 1px rgba(0, 0, 0, 0.1))',
      borderTop: '8px solid transparent',
      borderBottom: '8px solid transparent',
      borderRight: `8px solid ${theme.Tooltip.backgroundColor}`,
      top: '50%',
      transform: 'translateY(-50%)',
      left: -7,
    },
  },
  "&[data-position='bottom'] .tooltip-content": {
    top: '100%',
    marginTop: 10,
    left: '50%',
    transform: 'translateX(-50%)',
    '& .tooltip-arrow': {
      filter: 'drop-shadow(0px -2px 1px rgba(0, 0, 0, 0.1))',
      borderLeft: '12px solid transparent',
      borderRight: '12px solid transparent',
      borderBottom: `12px solid ${theme.Tooltip.backgroundColor}`,
      bottom: '90%',
      left: '50%',
      transform: 'translateX(-50%)',
    },
  },
  "&[data-position='left'] .tooltip-content": {
    right: '100%',
    marginRight: '8px !important',
    top: 0,
    bottom: 0,
    margin: 'auto 0',
    '& .tooltip-arrow': {
      filter: 'drop-shadow(2px 0px 1px rgba(0, 0, 0, 0.1))',
      borderTop: '8px solid transparent',
      borderBottom: '8px solid transparent',
      borderLeft: `8px solid ${theme.Tooltip.backgroundColor}`,
      top: '50%',
      transform: 'translateY(-50%)',
      right: -7,
    },
  },
});

const filterTooltipProps = (props: ToolTipProps) => filterProps(props, ['position', 'body', 'isOpen']);

export const Tooltip = ({children, ...props}: ToolTipProps) => {
  const filteredProps = filterTooltipProps(props);
  const {body, position = 'top', isOpen = true} = props;

  return (
    <div
      css={(theme: Theme) => tooltipStyle(theme)}
      data-position={position}
      {...filteredProps}
      data-testid="tooltip-wrapper"
    >
      {isOpen && (
        <div className="tooltip-content" data-testid="tooltip-content">
          {body}
          <div className="tooltip-arrow"></div>
        </div>
      )}
      {children}
    </div>
  );
};
