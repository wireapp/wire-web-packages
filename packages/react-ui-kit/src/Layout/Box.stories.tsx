/*
 * Wire
 * Copyright (C) 2025 Wire Swiss GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see http://www.gnu.org/licenses/.
 *
 */

import {Meta, StoryObj} from '@storybook/react';

import {Box} from './Box';

import {COLOR} from '../Identity/colors';
import {Text} from '../Text';

const meta: Meta<typeof Box> = {
  component: Box,
  title: 'Layout/Box',
  decorators: [
    Story => (
      <div style={{padding: '24px', maxWidth: '600px', margin: '0 auto'}}>
        <Story />
      </div>
    ),
  ],
};

export default meta;
type Story = StoryObj<typeof Box>;

const LOREM_IPSUM = `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce a mattis nibh, sed maximus leo. Fusce a lacinia sem,
vitae ornare dolor. Quisque rhoncus, magna non lacinia sagittis, erat augue fringilla metus, eu consectetur leo velit
non lacus. Phasellus ipsum turpis, dapibus ut purus in, lobortis consectetur mi.`;

export const Default: Story = {
  args: {
    children: LOREM_IPSUM,
  },
};

export const WithCustomPadding: Story = {
  args: {
    children: LOREM_IPSUM,
    style: {padding: '32px'},
  },
};

export const WithCustomBorder: Story = {
  args: {
    children: LOREM_IPSUM,
    style: {border: `2px solid ${COLOR.BLUE}`},
  },
};

export const WithFormattedContent: Story = {
  args: {
    children: (
      <>
        <Text block style={{fontSize: '20px', fontWeight: 'bold', marginBottom: '16px'}}>
          Important Notice
        </Text>
        <Text block style={{marginBottom: '12px'}}>
          {LOREM_IPSUM}
        </Text>
        <Text block style={{color: COLOR.GRAY}}>
          Last updated: 2025-01-01
        </Text>
      </>
    ),
  },
};

export const Nested: Story = {
  args: {
    children: (
      <>
        <Text block style={{marginBottom: '16px'}}>
          Outer box content
        </Text>
        <Box style={{backgroundColor: COLOR.GRAY_LIGHTEN_92}}>
          <Text>Inner box content</Text>
        </Box>
      </>
    ),
  },
};

export const Interactive: Story = {
  args: {
    children: (
      <div style={{textAlign: 'center'}}>
        <Text block style={{marginBottom: '16px'}}>
          Click the button below
        </Text>
        <button
          style={{
            backgroundColor: COLOR.BLUE,
            border: 'none',
            borderRadius: '4px',
            color: COLOR.WHITE,
            cursor: 'pointer',
            padding: '8px 16px',
          }}
        >
          Action Button
        </button>
      </div>
    ),
    style: {cursor: 'pointer'},
  },
};
