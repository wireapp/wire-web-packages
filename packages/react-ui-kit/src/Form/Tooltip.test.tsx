/*
 * Wire
 * Copyright (C) 2023 Wire Swiss GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see http://www.gnu.org/licenses/.
 *
 */

import * as React from 'react';

import '@testing-library/jest-dom/jest-globals';
import {render, screen} from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import {Tooltip} from './Tooltip';

import {StyledApp, THEME_ID} from '../Layout';

describe('<Tooltip />', () => {
  it('renders correctly', () => {
    const tree = render(
      <StyledApp themeId={THEME_ID.LIGHT}>
        <Tooltip body={<div>Tooltip Content</div>}>Hover Me</Tooltip>
      </StyledApp>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('hides tooltip content on mouse leave', () => {
    render(
      <StyledApp themeId={THEME_ID.LIGHT}>
        <Tooltip body="Tooltip Content">Hover Me</Tooltip>
      </StyledApp>,
    );
    const tooltipWrapper = screen.getByTestId('tooltip-wrapper');
    userEvent.hover(tooltipWrapper);
    userEvent.unhover(tooltipWrapper);
    expect(screen.getByTestId('tooltip-content')).not.toBeVisible();
  });

  it('should render tooltip on the top', () => {
    render(
      <StyledApp themeId={THEME_ID.LIGHT}>
        <Tooltip body="Tooltip Content" position="top">
          Hover Me
        </Tooltip>
      </StyledApp>,
    );
    expect(screen.getByTestId('tooltip-content')).toHaveStyle({bottom: '100%'});
  });

  it('should render tooltip on the right', () => {
    render(
      <StyledApp themeId={THEME_ID.LIGHT}>
        <Tooltip body="Tooltip Content" position="right">
          Hover Me
        </Tooltip>
      </StyledApp>,
    );
    expect(screen.getByTestId('tooltip-content')).toHaveStyle({left: '100%'});
  });

  it('should render tooltip on the bottom', () => {
    render(
      <StyledApp themeId={THEME_ID.LIGHT}>
        <Tooltip body="Tooltip Content" position="bottom">
          Hover Me
        </Tooltip>
      </StyledApp>,
    );
    expect(screen.getByTestId('tooltip-content')).toHaveStyle({top: '100%'});
  });

  it('should render tooltip on the left', () => {
    render(
      <StyledApp themeId={THEME_ID.LIGHT}>
        <Tooltip body="Tooltip Content" position="left">
          Hover Me
        </Tooltip>
      </StyledApp>,
    );
    expect(screen.getByTestId('tooltip-content')).toHaveStyle({right: '100%'});
  });
});
