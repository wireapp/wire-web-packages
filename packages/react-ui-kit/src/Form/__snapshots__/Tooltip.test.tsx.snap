// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tooltip /> renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  background: #edeff0;
  -webkit-transition: background 0.15s;
  transition: background 0.15s;
}

.emotion-1 {
  position: relative;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.emotion-1:hover .tooltip-content,
.emotion-1:focus-within .tooltip-content,
.emotion-1 {
  visibility: visible;
  opacity: 1;
}

.emotion-1 .tooltip-content {
  text-align: center;
  visibility: hidden;
  opacity: 0;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  height: -webkit-max-content;
  height: -moz-max-content;
  height: max-content;
  position: absolute;
  box-sizing: border-box;
  display: block;
  margin: 0 auto;
  padding: 4px 8px;
  background-color: #fff;
  color: rgb(29, 30, 31);
  border-radius: 4px;
  font-size: 0.75rem;
  font-weight: 500;
  white-space: nowrap;
  -webkit-transition: opacity ease-out 150ms,bottom ease-out 150ms;
  transition: opacity ease-out 150ms,bottom ease-out 150ms;
  box-shadow: 1px 2px 6px rgba(0, 0, 0, 0.3);
}

.emotion-1 .tooltip-content .tooltip-arrow {
  position: absolute;
  width: 0;
  height: 0;
}

.emotion-1[data-position='top'] .tooltip-content {
  bottom: 100%;
  margin-bottom: 10px;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -moz-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

.emotion-1[data-position='top'] .tooltip-content .tooltip-arrow {
  -webkit-filter: drop-shadow(0px 2px 1px rgba(0, 0, 0, 0.1));
  filter: drop-shadow(0px 2px 1px rgba(0, 0, 0, 0.1));
  border-left: 12px solid transparent;
  border-right: 12px solid transparent;
  border-top: 12px solid #fff;
  top: 90%;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -moz-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

.emotion-1[data-position='right'] .tooltip-content {
  left: 100%;
  margin-left: 8px!important;
  top: 0;
  bottom: 0;
  margin: auto 0;
}

.emotion-1[data-position='right'] .tooltip-content .tooltip-arrow {
  -webkit-filter: drop-shadow(-2px 0px 1px rgba(0, 0, 0, 0.1));
  filter: drop-shadow(-2px 0px 1px rgba(0, 0, 0, 0.1));
  border-top: 8px solid transparent;
  border-bottom: 8px solid transparent;
  border-right: 8px solid #fff;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -moz-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  left: -7px;
}

.emotion-1[data-position='bottom'] .tooltip-content {
  top: 100%;
  margin-top: 10px;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -moz-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

.emotion-1[data-position='bottom'] .tooltip-content .tooltip-arrow {
  -webkit-filter: drop-shadow(0px -2px 1px rgba(0, 0, 0, 0.1));
  filter: drop-shadow(0px -2px 1px rgba(0, 0, 0, 0.1));
  border-left: 12px solid transparent;
  border-right: 12px solid transparent;
  border-bottom: 12px solid #fff;
  bottom: 90%;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -moz-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

.emotion-1[data-position='left'] .tooltip-content {
  right: 100%;
  margin-right: 8px!important;
  top: 0;
  bottom: 0;
  margin: auto 0;
}

.emotion-1[data-position='left'] .tooltip-content .tooltip-arrow {
  -webkit-filter: drop-shadow(2px 0px 1px rgba(0, 0, 0, 0.1));
  filter: drop-shadow(2px 0px 1px rgba(0, 0, 0, 0.1));
  border-top: 8px solid transparent;
  border-bottom: 8px solid transparent;
  border-left: 8px solid #fff;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -moz-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  right: -7px;
}

<body>
    <div>
      <div
        class="emotion-0"
      >
        <div
          class="emotion-1"
          data-position="top"
          data-testid="tooltip-wrapper"
        >
          <div
            class="tooltip-content"
            data-testid="tooltip-content"
          >
            <div>
              Tooltip Content
            </div>
            <div
              class="tooltip-arrow"
            />
          </div>
          Hover Me
        </div>
      </div>
    </div>
  </body>,
  "container": .emotion-0 {
  background: #edeff0;
  -webkit-transition: background 0.15s;
  transition: background 0.15s;
}

.emotion-1 {
  position: relative;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.emotion-1:hover .tooltip-content,
.emotion-1:focus-within .tooltip-content,
.emotion-1 {
  visibility: visible;
  opacity: 1;
}

.emotion-1 .tooltip-content {
  text-align: center;
  visibility: hidden;
  opacity: 0;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  height: -webkit-max-content;
  height: -moz-max-content;
  height: max-content;
  position: absolute;
  box-sizing: border-box;
  display: block;
  margin: 0 auto;
  padding: 4px 8px;
  background-color: #fff;
  color: rgb(29, 30, 31);
  border-radius: 4px;
  font-size: 0.75rem;
  font-weight: 500;
  white-space: nowrap;
  -webkit-transition: opacity ease-out 150ms,bottom ease-out 150ms;
  transition: opacity ease-out 150ms,bottom ease-out 150ms;
  box-shadow: 1px 2px 6px rgba(0, 0, 0, 0.3);
}

.emotion-1 .tooltip-content .tooltip-arrow {
  position: absolute;
  width: 0;
  height: 0;
}

.emotion-1[data-position='top'] .tooltip-content {
  bottom: 100%;
  margin-bottom: 10px;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -moz-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

.emotion-1[data-position='top'] .tooltip-content .tooltip-arrow {
  -webkit-filter: drop-shadow(0px 2px 1px rgba(0, 0, 0, 0.1));
  filter: drop-shadow(0px 2px 1px rgba(0, 0, 0, 0.1));
  border-left: 12px solid transparent;
  border-right: 12px solid transparent;
  border-top: 12px solid #fff;
  top: 90%;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -moz-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

.emotion-1[data-position='right'] .tooltip-content {
  left: 100%;
  margin-left: 8px!important;
  top: 0;
  bottom: 0;
  margin: auto 0;
}

.emotion-1[data-position='right'] .tooltip-content .tooltip-arrow {
  -webkit-filter: drop-shadow(-2px 0px 1px rgba(0, 0, 0, 0.1));
  filter: drop-shadow(-2px 0px 1px rgba(0, 0, 0, 0.1));
  border-top: 8px solid transparent;
  border-bottom: 8px solid transparent;
  border-right: 8px solid #fff;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -moz-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  left: -7px;
}

.emotion-1[data-position='bottom'] .tooltip-content {
  top: 100%;
  margin-top: 10px;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -moz-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

.emotion-1[data-position='bottom'] .tooltip-content .tooltip-arrow {
  -webkit-filter: drop-shadow(0px -2px 1px rgba(0, 0, 0, 0.1));
  filter: drop-shadow(0px -2px 1px rgba(0, 0, 0, 0.1));
  border-left: 12px solid transparent;
  border-right: 12px solid transparent;
  border-bottom: 12px solid #fff;
  bottom: 90%;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -moz-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

.emotion-1[data-position='left'] .tooltip-content {
  right: 100%;
  margin-right: 8px!important;
  top: 0;
  bottom: 0;
  margin: auto 0;
}

.emotion-1[data-position='left'] .tooltip-content .tooltip-arrow {
  -webkit-filter: drop-shadow(2px 0px 1px rgba(0, 0, 0, 0.1));
  filter: drop-shadow(2px 0px 1px rgba(0, 0, 0, 0.1));
  border-top: 8px solid transparent;
  border-bottom: 8px solid transparent;
  border-left: 8px solid #fff;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -moz-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  right: -7px;
}

<div>
    <div
      class="emotion-0"
    >
      <div
        class="emotion-1"
        data-position="top"
        data-testid="tooltip-wrapper"
      >
        <div
          class="tooltip-content"
          data-testid="tooltip-content"
        >
          <div>
            Tooltip Content
          </div>
          <div
            class="tooltip-arrow"
          />
        </div>
        Hover Me
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
